File: client/lib/models/patient_item.dart
class PatientItem {
  final int id;
  final bool active;
  final String firstName;
  final String fullName;
  final DateTime? dateOfBirth;
  PatientItem({
    required this.id,
    required this.active,
    required this.firstName,
    required this.fullName,
    this.dateOfBirth,
  });

  factory PatientItem.fromJson(Map<String, dynamic> json) {
    return PatientItem(
      id: json['id'],
      active: json['active'],
      firstName: json['first_name'],
      fullName: json['full_name'],
      dateOfBirth: json['date_of_birth'] != null
          ? DateTime.parse(json['date_of_birth'])
          : null,
    );
  }

  String initials() {
    return firstName[0] + fullName[0];
  }
}

File: server/go/models/patient_item.go
package models

import "time"

type PatientItem struct {
	ID          uint       `gorm:"primaryKey" json:"id"`
	Active      bool       `json:"active"`
	FirstName   string     `gorm:"size:50" json:"first_name"`
	FullName    string     `gorm:"size:50" json:"full_name"`
	DateOfBirth *time.Time `json:"date_of_birth"`
}
